x/lockup/keeper/msg_server_test.go

```diff
@@ -9,55 +9,97 @@ import (
-	msgServer := keeper.NewMsgServerImpl(*suite.app.LockupKeeper)
-	_, err = msgServer.LockTokens(sdk.WrapSDKContext(suite.ctx), types.NewMsgLockTokens(addr1, time.Second, coins))
-
-	// check locks
-	locks, err := suite.app.LockupKeeper.GetPeriodLocks(suite.ctx)
-	suite.Require().NoError(err)
-	suite.Require().Len(locks, 1)
-	suite.Require().Equal(locks[0].Coins, coins)
-	// check accumulation store is correctly updated
-	accum := suite.app.LockupKeeper.GetPeriodLocksAccumulation(suite.ctx, types.QueryCondition{
-		LockQueryType: types.ByDuration,
-		Denom:         "stake",
-		Duration:      time.Second,
-	})
-	suite.Require().Equal(accum.String(), "10")
-
-	// add more tokens to lock via LockTokens
-	addCoins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
-	err = simapp.FundAccount(suite.app.BankKeeper, suite.ctx, addr1, addCoins)
-	suite.Require().NoError(err)
-
-	_, err = msgServer.LockTokens(sdk.WrapSDKContext(suite.ctx), types.NewMsgLockTokens(addr1, locks[0].Duration, addCoins))
-	suite.Require().NoError(err)
-
-	// check locks after adding tokens to lock
-	locks, err = suite.app.LockupKeeper.GetPeriodLocks(suite.ctx)
-	suite.Require().NoError(err)
-	suite.Require().Len(locks, 1)
-	suite.Require().Equal(locks[0].Coins, coins.Add(addCoins...))
-
-	// check accumulation store is correctly updated
-	accum = suite.app.LockupKeeper.GetPeriodLocksAccumulation(suite.ctx, types.QueryCondition{
-		LockQueryType: types.ByDuration,
-		Denom:         "stake",
-		Duration:      time.Second,
-	})
-	suite.Require().Equal(accum.String(), "20")
+	type param struct {
+		coinsToLock         sdk.Coins
+		lockOwner           sdk.AccAddress
+		duration            time.Duration
+		coinsInOwnerAddress sdk.Coins
+	}
+
+	tests := []struct {
+		name       string
+		param      param
+		expectPass bool
+	}{
+		{
+			name: "creation of lock via lockTokens",
+			param: param{
+				coinsToLock:         sdk.Coins{sdk.NewInt64Coin("stake", 10)},       // setup wallet
+				lockOwner:           sdk.AccAddress([]byte("addr1---------------")), // setup wallet
+				duration:            time.Second,
+				coinsInOwnerAddress: sdk.Coins{sdk.NewInt64Coin("stake", 10)},
+			},
+			expectPass: true,
+		},
+		{
+			name: "locking more coins than are in the address",
+			param: param{
+				coinsToLock:         sdk.Coins{sdk.NewInt64Coin("stake", 20)},       // setup wallet
+				lockOwner:           sdk.AccAddress([]byte("addr1---------------")), // setup wallet
+				duration:            time.Second,
+				coinsInOwnerAddress: sdk.Coins{sdk.NewInt64Coin("stake", 10)},
+			},
+			expectPass: false,
+		},
+	}
+
+	for _, test := range tests {
+		suite.SetupTest()
+
+		if test.expectPass {
+			// Setup the Account
+			err := simapp.FundAccount(suite.app.BankKeeper, suite.ctx, test.param.lockOwner, test.param.coinsInOwnerAddress)
+			suite.Require().NoError(err)
+
+			_, err = suite.app.LockupKeeper.LockTokens(suite.ctx, test.param.lockOwner, test.param.coinsToLock, test.param.duration)
```

Comment:
This structure isn't quite right.

It should do the funding of the account & the lock tokens off the params. Then it should do the checks based on the `if expPass` logic. 

E.g. imagine making this scalable, for adding more test cases with coinsToLock having `100k uosmo`, or coinsToLock having multiple entries `100uosmo,10uatom`

Reply:
In that case should I use `addr1` or `lockOwner` inside `if expPass` logic to test `LockTokens`?

My guess is to use `addr1` because it stays consistent across tests but then why would we even need `lockOwner` in params?

Reply:
Use `lockowner`, thats why its a param!

Reply:
In that case, I think we can still place `LockTokens` outside the `if expectPass / else` logic and only check if requires error or not within the if/else.

Reply:
Fixed!